drift-transport-netty/src/main/java/com/facebook/drift/transport/netty/client/DriftNettyMethodInvokerFactory.java

```diff
@@ -92,38 +98,50 @@ public DriftNettyMethodInvokerFactory(
         this.defaultSocksProxy = Optional.ofNullable(factoryConfig.getSocksProxy());
 
         connectionPoolMaintenanceExecutor = newSingleThreadScheduledExecutor(daemonThreadsNamed("drift-connection-maintenance"));
-
-        ConnectionManager connectionManager = new ConnectionFactory(group, sslContextFactory, allocator, factoryConfig);
-        if (factoryConfig.isConnectionPoolEnabled()) {
-            connectionManager = new ConnectionPool(
-                    connectionManager,
-                    group,
-                    factoryConfig.getConnectionPoolMaxSize(),
-                    factoryConfig.getConnectionPoolMaxConnectionsPerDestination(),
-                    factoryConfig.getConnectionPoolIdleTimeout(),
-                    connectionPoolMaintenanceExecutor);
-        }
-        this.connectionManager = connectionManager;
+        connectionFactory = new ConnectionFactory(group, sslContextFactory, allocator, factoryConfig);
     }
 
     @Override
     public MethodInvoker createMethodInvoker(I clientIdentity)
     {
-        ConnectionParameters clientConfig = toConnectionConfig(clientConfigurationProvider.apply(clientIdentity));
+        DriftNettyClientConfig driftNettyClientConfig = clientConfigurationProvider.apply(clientIdentity);
+        ConnectionParameters clientConfig = toConnectionConfig(driftNettyClientConfig);
 
         // validate ssl context configuration is valid
         clientConfig.getSslContextParameters()
                 .ifPresent(sslContextParameters -> sslContextFactory.get(sslContextParameters).get());
 
+        ConnectionManager connectionManager = getConnectionManager(clientIdentity, driftNettyClientConfig);
         return new DriftNettyMethodInvoker(clientConfig, connectionManager, group);
     }
 
+    public ConnectionManager getConnectionManager(I clientIdentity, DriftNettyClientConfig driftNettyClientConfig)
+    {
+        boolean connectionPoolEnabled = firstNonNull(driftNettyClientConfig.getConnectionPoolEnabled(), factoryConfig.isConnectionPoolEnabled());
+        if (!connectionPoolEnabled) {
+            return connectionFactory;
+        }
+
+        int connectionPoolMaxSize = firstNonNull(driftNettyClientConfig.getConnectionPoolMaxSize(), factoryConfig.getConnectionPoolMaxSize());
```

Comment:
There can be interesting "mixed behavior" when , for example:
- `driftNettyClientConfig` set `connectionPoolMaxSize`, but doesn't set `maxConnectionsPerDestination`
- So `connectionPoolMaxSize` will be from `driftNettyClientConfig`, while `maxConnectionsPerDestination` is from `factoryConfig `.

This is fine. But I am thinking maybe if `connectionPoolEnabled` is set to true from `driftNettyClientConfig`, we also ask `driftNettyClientConfig` to set other values such as `connectionPoolMaxSize` and `maxConnectionsPerDestination` ?

Reply:
I was thinking about more property - by - property overrides. So basically you can set some defaults, and then override them on per property basis per client.